lib/checks/aria/unsupportedattr.js

```diff
@@ -0,0 +1,22 @@
+let unsupported;
+if (node.hasAttributes()) {
+	unsupported = [].slice
+		.call(node.attributes)
+		.filter(candidate => {
+			// skip non-ARIA attributes
+			if (/aria-/.test(candidate.name)) {
+				// filter out unsupported attributes
+				return !axe.commons.aria.validateAttr(candidate.name, {
+					flagUnsupported: true
+				});
+			}
+		})
+		.map(candidate => {
+			return candidate.name.toString();
+		});
+}
+if (unsupported.length) {
+	this.data(unsupported);
+	return true;
+}
+return false;
```

Comment:
I'm not sure the logic works here. Shouldn't you return `true` when this passes? I would also think that this needs to be in `all` instead of `any` in the rule. That decision may be why you  used this strange logic.

Reply:
@WilcoFiers the `aria-roles` unsupported stuff works on `none`. I would actually think that should be the case here as well.